### ISSUE TITLE
Self-Rating for books in the app

### ISSUE THREAD

**Issue by raviwarrier** [advplyr/audiobookshelf-app#236]
### Is your feature request related to a problem? Please describe.
It's not a problem that its not there, it's just a good to have feature. Upon completing a book (or marking it as complete), if one was allowed to give a rating (5 stars in increments of 1/2 star or even 1 star), it would be nice. Especially if the library is shared with other users who might be interested in what others thought of the book.

### Describe the solution you'd like
A 5-star rating system for each book upon completion (playback complete) or marking the book as complete. With multiple users, an average of the rating is displayed in the book item page for others to view as they contemplate/consider reading the book.

### Describe alternatives you've considered
None.

**Comment by HeroYui** [comment]
If this is implemented I think it is worth to at least think about interfacing other services like Goodreads for this. It is not necessary in the beginning but keeping the implementation open to this would be SOLID.

**Comment by drysdan** [comment]
A 5 star rating system would be a lovely addition!  Also the ability for each user to append a brief comment or review blurb to a book or series, so other users of the library can get recommendations

**Comment by AdamCerutti** [comment]
I'd also love to see this feature! It would allow users to filter or sort based on their own ratings!

**Comment by jazzgott** [comment]
Yes please! I'd love this to be implemented. I would not bother with goodreads unless you plan to add a similar site for all possible languages (in that case add scrapers for services in different languages too, perhaps as third party plugin, just as stash does. I'm sure you know what stash is...). I know who's got a similar rating system already implemented. You guessed it :) it's stash.

**Comment by ulimanaio** [comment]
Yes, a rating would be amazing. Maybe similar to audible 5 stars for
Story and author

**Comment by rohit-purandare** [comment]
Also voting for this! I'd love to pull it in via API with [Shelfbridge](https://github.com/rohit-purandare/ShelfBridge) and send to Hardcover so it can generate recommendations for users.

**Comment by jaroslawhartman** [comment]
Still evaluating ABS but this may be a single dealbreaker for me. I'd love to bring more order to my library and a rating system (combined with an external metadata scraper to pull rating from the external source) is one of my must-have.

**Comment by SeiferVIII** [comment]
Strong Point, I love this feature, maybe add the 5 star rading or 100 points system or decimal system like 7.8 etc.
Thank you again for the time and ideas

**Comment by sytar** [comment]
Overall I love AudioBookShelf, but was surprised when there wasn't a way to rank the books.  I do hope that this gets added.  Integration with GoodReads or something would be nice, but even just a basic ranking system of 1-5 stars would be nice.

**Comment by fangar** [comment]
Add me to the list of users that would find this beneficial. Yes I can leave a review on GR or a word doc or something. But I'd rather do it on ABS. As refined as ABS is, I was shocked that it's not already a thing.

**Comment by peeceejones** [comment]
Just arrived here from the ABD Reddit where I searched for this feature. I just finished a book - and wanted to rate it - to find thats not in the software (yet?). Would love a way to rank my listens, and see / recommend to my other users - in the same way Plex does.

**Comment by andremlk** [comment]
Please add this

**Comment by raviwarrier** [comment]
@advplyr and team, I raised this request over 4 years ago. It's a good feature to have and useful and wanted as seen by the many comments. 

Please consider doing this now that the app is much more stable than it was even I first asked for it.

**Comment by christoph288** [comment]
I would also love to have this useful feature.
